common.h

```diff
@@ -13,7 +13,7 @@ template <typename Base> class Identifiable : public Base {
 	}
 
 private:
-	const Id m_id;
+	Id m_id;

```

Comment:
With the changes I described in the first comment, we may be able to put the const back on here!

Reply:
I forgot about it before making the commit, but it can't be added yet anyway because my method of replacing elements still doesn't seem compatible with it.